src/jit/target.h

```diff
@@ -1529,7 +1529,7 @@ typedef unsigned short regPairNoSmall; // arm: need 12 bits
   #define FEATURE_MULTIREG_STRUCT_PROMOTE 1  // True when we want to promote fields of a multireg struct into registers
   #define FEATURE_FASTTAILCALL     1       // Tail calls made as epilog+jmp
   #define FEATURE_TAILCALL_OPT     1       // opportunistic Tail calls (i.e. without ".tail" prefix) made as fast tail calls.
-  #define FEATURE_SET_FLAGS        1       // Set to true to force the JIT to mark the trees with GTF_SET_FLAGS when the flags need to be set
+  #define FEATURE_SET_FLAGS        0       // Set to true to force the JIT to mark the trees with GTF_SET_FLAGS when the flags need to be set
```

Comment:
We are using XARCH solution not ARM solution

Reply:
To me, this comment just adds to the confusion. AFAICT, this behavior is, in fact, required, but in the `!LEGACY_BACKEND && !FEATURE_SET_FLAGS` path vs. the `LEGACY_BACKEND && FEATURE_SET_FLAGS` path.
Am I missing something here?

Reply:
FEATURE_SET_FLAGS bring is a lot of code which is designed and tested with Legacy JIT and ARM32.  Since XARCH has this feature disabled but has functionally equivalent code working with RyuJIT, I took the approach to follow XARCH/RyuJIT pattern.

Reply:
Maybe we should rename `FEATURE_SET_FLAGS` to `FEATURE_LEGACY_SET_FLAGS`

Reply:
Do you want it this change or a separate change.

Reply:
Definitely a separate change, or just open an issue for someone else to do

Reply:
#14175